manifests/database/postgresql.pp

```diff
@@ -10,20 +13,111 @@
     default => postgresql_password($::foreman::db_username, $::foreman::db_password),
   }
 
+  $rep_password = $::foreman::db_replication_password ? {
+    'UNSET' => false,
+    default => postgresql_password($::foreman::db_replication_username, $::foreman::db_replication_password),
+  }
+
+  $port = $::foreman::db_port ? {
+    'UNSET' => 5432,
+    default => $::foreman::db_port,
+  }
+
   # Prevents errors if run from /root etc.
   Postgresql_psql {
     cwd => '/',
   }
 
-  include ::postgresql::client, ::postgresql::server
+  if $::foreman::db_cluster {
+    $db_all_cluster_hostnames = concat($::foreman::db_cluster_hostnames, $::fqdn)
+    if $::foreman::db_node_type == 'master' {
+      class {'postgresql::server':
+        listen_addresses => '*',
+      }
+      if $::foreman::db_synchronous_names {
+        postgresql::server::config_entry { 'synchronous_standby_names':
+          value => join($::foreman::db_synchronous_names, ","),
+        }
+      }
+      postgresql::server::db { $dbname:
+        user     => $::foreman::db_username,
+        password => $password,
+        owner    => $::foreman::db_username,
+        encoding => 'utf8',
+        locale   => 'en_US.utf8',
+      }
+    } elsif $::foreman::db_node_type == 'slave' {
+      file { 'Write .pgpass':
+        content => "*:*:*:${::foreman::db_replication_username}:${::foreman::db_replication_password}\n",
+        path    => "${::foreman::params::postgres_home}/.pgpass",
```

Comment:
It surprises me a little that there's no variable for this in puppetlabs-postgresql.